### ISSUE TITLE
Executable returns OutOfMemory on `std.process.argsFree` if an error is returned

### ISSUE THREAD

**Issue by iansimonson** [ziglang/zig#6379]
*Edit*: Edited title to be more accurate

When returning an error from main which has processed command line arguments, the error return trace is strange / incorrect. This seems to only happen when processing command line arguments (as in allocating memory in general doesn't cause this to occur) and only on the error path. On success the OOM doesn't appear.

It also only occurs if at least one command argument is long enough.

Basic example:

```zig
const std = @import("std");

var gpa = std.heap.GeneralPurposeAllocator(.{}){};
const alloc = &gpa.allocator;

// also happens with the c_allocator
//const alloc = std.heap.c_allocator;

pub fn main() !void {
    defer _ = gpa.deinit();
    const args = try std.process.argsAlloc(alloc);
    defer std.process.argsFree(alloc, args);
    return error.some_error;
}
```

Input:
```
./oom abcdefghi
```
Outputs:
```
error: some_error
/Users/ian/local/lib/zig/std/heap/general_purpose_allocator.zig:562:13: 0x10b08c147 in std.heap.general_purpose_allocator.GeneralPurposeAllocator((struct std.heap.general_purpose_allocator.Config constant)).resize (oom)
            return error.OutOfMemory;
            ^
/Users/ian/development/zig/single_files/oom.zig:12:5: 0x10b08c8e6 in main (oom)
    return error.some_error;
    ^
```

HOWEVER
Input:
```
./oom a
```
Outputs:
```
error: some_error
/Users/ian/development/zig/single_files/oom.zig:12:5: 0x1093fb8e6 in main (oom)
    return error.some_error;
    ^
```

**Comment by LemonBoy** [comment]
Duplicate of #1923, the `OutOfMemory` error was thrown and handled somewhere else in the allocator interface and you shouldn't be seeing it.

TLDR: There's no OOM

**Comment by iansimonson** [comment]
Yep looks like it. The following even simpler example shows the behavior:

```zig
fn foo() !void {
    return error.an_error;
}
pub fn main() !void {
    _ = foo() catch {};
    return error.some_error;
}
```

prints:
```
error: some_error
/Users/ian/development/zig/single_files/oom.zig:2:5: 0x10575c8e2 in foo (oom)
    return error.an_error;
    ^
/Users/ian/development/zig/single_files/oom.zig:6:5: 0x10575c8bd in main (oom)
    return error.some_error;
    ^
```
